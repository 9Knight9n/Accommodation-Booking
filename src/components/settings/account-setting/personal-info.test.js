import React from 'react';
import ReactDom from 'react-dom';
import {render, fireEvent, cleanup, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PersonalInfo from '../account-setting/personal-info';
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer';
afterEach(cleanup)

it('profile renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<PersonalInfo/>, div);
  });
  

// it("email contains @", () =>{
//     const {getByTestId} = render(<PersonalInfo />)
//     expect(getByTestId("personalInfo-emailId").
// })

it("email is not null", () =>{
    const {getByTestId} = render(<PersonalInfo />);
    expect(getByTestId("personalInfo-emailId")).not.toBeNull;
})

it("firstName is not null", () =>{
    const {getByTestId} = render(<PersonalInfo />);
    expect(getByTestId("personalInfo-firstName")).not.toBeNull;
})

it("lastName is not null", () =>{
    const {getByTestId} = render(<PersonalInfo />);
    expect(getByTestId("personalInfo-lastName")).not.toBeNull;
})

test('national code must at least 10 digits', () =>{
    render(<PersonalInfo />);
    const inputNationalCode = screen.getByTestId('personalInfo-nationalId');
    userEvent.type(inputNationalCode, '12345');
    expect(inputNationalCode).not.toHaveValue('12345');
});

test('email must be valid', () =>{
    render(<PersonalInfo />);
    const inputEmailId= screen.getByTestId('personalInfo-emailId');
    userEvent.type(inputEmailId, 'abdskdw.321dmwi.com');
    expect(inputEmailId).not.toHaveValue('abdskdw.321dmwi.com');
});

// it("matches snapShot", () =>{
//     const tree = renderer.create(<PersonalInfo />).toJSON();
//     expect(tree).toMatchSnapshot();
// })

