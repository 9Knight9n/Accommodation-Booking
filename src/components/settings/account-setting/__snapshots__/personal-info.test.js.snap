// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="personalInfo-main"
>
  <div
    className="Toastify"
  />
  <div
    className="personalInfo-avatar mt-4 mb-4"
  >
    <div>
      <svg
        fill="currentColor"
        height={100}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "black",
          }
        }
        viewBox="0 0 16 16"
        width={100}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.468 12.37C12.758 11.226 11.195 10 8 10s-4.757 1.225-5.468 2.37A6.987 6.987 0 008 15a6.987 6.987 0 005.468-2.63z"
        />
        <path
          clipRule="evenodd"
          d="M8 9a3 3 0 100-6 3 3 0 000 6z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M8 1a7 7 0 100 14A7 7 0 008 1zM0 8a8 8 0 1116 0A8 8 0 010 8z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <form
    className="row"
  >
    <div
      className="personalInfo-form w-100 ml-5 mt-4"
    >
      <div
        className="firstName row"
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-firstName"
        >
          First Name:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control shadow-none form-control"
              data-testid="personalInfo-firstName"
              id="personalInfo-firstName"
              name="firstName"
              onChange={[Function]}
              type="text"
            />
            <div
              className="ml-1 invalid-feedback"
            >
              You must enter your first name!
            </div>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-lastName row"
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-lastName"
        >
          Last Name:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control shadow-none form-control"
              data-testid="personalInfo-lastName"
              id="personalInfo-lastName"
              name="lastName"
              onChange={[Function]}
              type="text"
            />
            <div
              className="ml-1 invalid-feedback"
            >
              You must enter your last name!
            </div>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2 "
      />
      <div
        className="personalInfo-nationalId row"
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-nationalId"
        >
          National Code:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control shadow-none form-control"
              data-testid="personalInfo-nationalId"
              id="personalInfo-nationalId"
              name="nationalId"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="ml-1 invalid-feedback"
            >
              National code must be up to 10 digits.
            </div>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-Gender "
      >
        <div
          className="row"
        >
          <label
            className="form-label col-lg-2 col-md-2 col-sm-3"
            htmlFor="personalInfo-Gender"
          >
            Gender:
          </label>
          <div
            className="form-select form-group col-lg-10 col-md-10 col-sm-9"
          >
            <select
              className=""
              data-testid="personalInfo-Gender"
              id="personalInfo-Gender"
              name="gender"
              onChange={[Function]}
              required={true}
            >
              <option>
                Male
              </option>
              <option>
                Female
              </option>
            </select>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-dateOfBirth row"
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-dateOfBirth"
        >
          Date Of Birth:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="MuiFormControl-root MuiTextField-root dateOfBirth-field"
            data-testid="personalInfo-dateOfBirth"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                defaultValue="2021-03-24"
                disabled={false}
                id="personalInfo-dateOfBirth"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="date"
              />
            </div>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-phoneNum row"
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-phoneNum"
        >
          Phone Number:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="input-group"
          >
            <div
              className="input-group-prepend"
              style={
                Object {
                  "width": "inherit",
                }
              }
            >
              <span
                className="input-group-btn"
              >
                <div
                  className=" react-tel-input"
                  onKeyDown={[Function]}
                  style={Object {}}
                >
                  <div
                    className="special-label"
                  >
                    Phone
                  </div>
                  <input
                    className="form-control shadow-none"
                    disabled={false}
                    id="personalInfo-phoneNum"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onCopy={[Function]}
                    onDoubleClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="1 (702) 123-4567"
                    style={
                      Object {
                        "width": "inherit",
                      }
                    }
                    type="phone-number"
                    value=""
                  />
                  <div
                    className=" flag-dropdown"
                    style={Object {}}
                  >
                    <div
                      aria-haspopup="listbox"
                      className="selected-flag"
                      onClick={[Function]}
                      role="button"
                      tabIndex="0"
                      title=""
                    >
                      <div
                        className="flag 0"
                      >
                        <div
                          className="arrow"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </span>
            </div>
            
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-emailId row "
      >
        <label
          className="form-label col-lg-2 col-md-2 col-sm-3"
          htmlFor="personalInfo-emailId"
        >
          Email:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <div
            className="input-group"
          >
            <input
              className="form-control shadow-none form-control"
              data-testid="personalInfo-emailId"
              disabled={true}
              id="personalInfo-emailId"
              name="email"
              onChange={[Function]}
              type="text"
            />
            <div
              className="ml-1 invalid-feedback"
            >
              Email is invalid!
            </div>
          </div>
        </div>
      </div>
      <hr
        className="personalInfo-line p-2"
      />
      <div
        className="personalInfo-bioField row"
      >
        <label
          className="form-lable col-lg-2 col-md-2 col-sm-3"
          htmlFor="bio"
        >
          Bio:
        </label>
        <div
          className="form-group col-lg-10 col-md-10 col-sm-9"
        >
          <textarea
            className="bio form-control"
            data-testid="personalInfo-ptextArea"
            id="ptextArea"
            maxLength="210"
            name="bio"
            onChange={[Function]}
          />
          <div
            className="personalInfo-textAreaCounter"
          >
            0
             of 210
          </div>
        </div>
      </div>
      <div
        className="w-100 d-flex justify-content-center mb-5"
      >
        <div
          className="personalInfo-btn mr-5 mb-2 mt-5"
        >
          <div
            className="mb-4"
          >
            <button
              className="btn btn-primary btn-lg btn-block"
              data-testid="personalInfo-submit"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </div>
          <div
            className="mb-4"
          >
            <button
              className="btn btn-secondary btn-lg btn-block"
              data-testid="personalInfo-reset"
              onClick={[Function]}
              type="button"
            >
              Reset
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
